storm-core/src/clj/org/apache/storm/daemon/nimbus.clj

```diff
@@ -1355,48 +1366,843 @@
 (defmethod blob-sync :local [conf nimbus]
   nil)
 
-(defserverfn service-handler [conf inimbus]
-  (.prepare inimbus conf (master-inimbus-dir conf))
-  (log-message "Starting Nimbus with conf " conf)
-  (let [nimbus (nimbus-data conf inimbus)
-        blob-store (:blob-store nimbus)
-        principal-to-local (AuthUtils/GetPrincipalToLocalPlugin conf)
+(defn extract-cluster-metrics [^ClusterSummary summ]
```

Comment:
Thinking it once more, we're in progress of porting so just referring unported thing is better than refactor.
I'll make this method to refer ui function to get rid of duplication.